frontend/src/pages/associations/files/Upload.tsx

```diff
@@ -0,0 +1,229 @@
+        initialValues: file,
+        onSubmit: values => {
+            newToast({
+                message: "Sauvegarde en cours",
+                level: ToastLevel.Success
+            });
+            api.files
+                .patch({
+                    id: values.id,
+                    name: values.name,
+                    description: values.description
+                })
+                .then(res => {
+                    newToast({
+                        message: "Fichier sauvegardé !",
+                        level: ToastLevel.Success
+                    });
+                })
+                .catch(err => {
+                    newToast({
+                        message: err.message,
+                        level: ToastLevel.Error
+                    });
+                });
+        }
+    });
+
+    let details;
+    if (!isCollapsed) {
+        details = (
+            <div>
+                <textarea
+                    id={"description"}
+                    name={"description"}
+                    className={"form-control border-0"}
+                    placeholder={"Description"}
+                    onChange={formik.handleChange}
+                    value={formik.values.description}
+                />
+                <Button
+                    size="sm"
+                    variant="success"
+                    style={{
+                        position: "absolute",
+                        bottom: "10px",
+                        right: "10px",
+                        width: "100px"
+                    }}
+                    type="submit"
+                >
+                    Sauvegarder
+                </Button>
+            </div>
+        );
+    }
+
+    return (
+        <form onSubmit={formik.handleSubmit}>
+            <Card className={"mt-3"}>
+                <Card.Header>
+                    ✅{" "}
+                    <Form.Control
+                        id="name"
+                        name="name"
+                        type="text"
+                        className={"border-0"}
+                        placeholder="Nom du fichier"
+                        onChange={formik.handleChange}
+                        value={formik.values.name}
+                    />
+                    <div className={"card-options"}>
+                        {isCollapsed ? (
+                            <i
+                                className="fe fe-chevron-down"
+                                onClick={() => setIsCollapsed(false)}
+                            />
+                        ) : (
+                            <i
+                                className="fe fe-chevron-up"
+                                onClick={() => setIsCollapsed(true)}
+                            />
+                        )}
+                    </div>
+                </Card.Header>
+                {!isCollapsed ? details : null}
+            </Card>
+        </form>
+    );
+};
```

Comment:
Lacks comments. Even one comment on top of every component to understand what it does and where it is used would be very useful.
For instance, why would `FileUploadDone` use a form? This is neither clear nor obvious.